opal/mca/memory/patcher/configure.m4

```diff
@@ -38,53 +38,18 @@ AC_DEFUN([MCA_opal_memory_patcher_CONFIG],[
 
     OPAL_VAR_SCOPE_PUSH([memory_patcher_have___curbrk memory_patcher_have___mmap memory_patcher_have___syscall memory_patcher_have___mmap_prototype memory_patcher_have___syscall_prototype])
 
-    memory_patcher_have___curbrk=0
-    memory_patcher_have___mmap=0
-    memory_patcher_have___mmap_prototype=0
-    memory_patcher_have___syscall=0
-    memory_patcher_have___syscall_prototype=0
+    AC_CHECK_FUNCS([__curbrk])
 
-    AC_MSG_CHECKING([for __curbrk symbol])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[char *tmp = __curbrk;])],
-                   [AC_MSG_RESULT([yes])
-                    memory_patcher_have___curbrk=1],
-                   [AC_MSG_RESULT([no])])
-    AC_DEFINE_UNQUOTED([OPAL_MEMORY_PATCHER_HAVE___CURBRK], [$memory_patcher_have___curbrk],
-                       [Whether the glibc __curbrk exists])
+    AC_CHECK_HEADERS([linux/mman.h sys/syscall.h])
 
-    AC_MSG_CHECKING([whether __mmap prototype exists])
-    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <sys/mman.h>],[char *tmp = __mmap (NULL, 0, 0, 0, 0, 0);])],
-                   [AC_MSG_RESULT([yes])
-                    memory_patcher_have___mmap_prototype=1],
-                   [AC_MSG_RESULT([no])])
-    AC_DEFINE_UNQUOTED([OPAL_MEMORY_PATCHER_HAVE___MMAP_PROTO], [$memory_patcher_have___mmap_prototype],
-                       [Whether the internal __mmap call has a prototype])
+    AC_CHECK_DECLS([__mmap], [], [], [#include <sys/mman.h>])
```

Comment:
I notice that you use `<sys/mman.h>` here, but check for `<linux/mman.h>`, above.  Was that intentional?

Did you want to make this test somehow dependent upon the result of the prior `AC_CHECK_HEADERS` result for `mman.h`?